packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -18,7 +22,38 @@ import {
   HttpVerb,
   Visibility,
 } from "@typespec/http";
-import { TCGCContext } from "./internal-utils.js";
+import { TspLiteralType } from "./internal-utils.js";
+
+export interface TCGCContext {
```

Comment:
is there too many props here? I don't see them set in the createTCGCcontecxt?

Reply:
it currently is only adding the required props, i have optional props that are more for creating the sdk package, so I'm trying to have `createTcgcContext` create the minimal interface needed

Reply:
but wouldn't you want that on the `SDKContext` then?

Reply:
ok that's fair, let me go ahead and do that. thanks!